frontend/modules/shared/ui/src/lib/layout/language-switcher/language-switcher.ts

```diff
@@ -0,0 +1,48 @@
+import { CommonModule } from '@angular/common';
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { FormControl, FormsModule, ReactiveFormsModule } from '@angular/forms';
+import { LangChangeEvent, TranslateService } from '@ngx-translate/core';
+import {
+  TuiButtonModule,
+  TuiDataListModule,
+  TuiFlagPipeModule,
+  TuiTextfieldControllerModule,
+} from '@taiga-ui/core';
+import { TuiCountryIsoCode, TuiLanguageName } from '@taiga-ui/i18n';
+import { TuiSelectModule } from '@taiga-ui/kit';
+
+@Component({
+  selector: 'reunice-language-switcher',
+  templateUrl: './language-switcher.html',
+  styleUrls: ['./language-switcher.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    CommonModule,
+    TuiButtonModule,
+    TuiDataListModule,
+    FormsModule,
+    ReactiveFormsModule,
+    TuiSelectModule,
+    TuiFlagPipeModule,
+    TuiTextfieldControllerModule,
+  ],
+})
+export class LanguageSwitcherComponent {
+  readonly translateService = inject(TranslateService);
+
+  readonly language = new FormControl(this.translateService.defaultLang);
+
+  readonly flags = new Map<TuiLanguageName, TuiCountryIsoCode>([
+    ['pl', TuiCountryIsoCode.PL],
+    ['en', TuiCountryIsoCode.GB],
+  ]);
+
+  readonly names: TuiLanguageName[] = Array.from(this.flags.keys());
+
+  constructor() {
+    this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
```

Comment:
I'm certain this subscribe should be in some other file but found no way how to do it XD
Also we should change the html lang tag along with it but it seems like the ngx-translate doesn't have any functionality like that?

Reply:
the lang tag thingie not possible without a hack https://github.com/ngx-translate/core/issues/933

Reply:
Since there should be only one language switcher, you can leave the subscription here. It's not the best place, but I wouldn't bother with it for now.

You can change html lang attribute with this:
```
constructor(@Inject(DOCUMENT) document: Document) {
  document.documentElement.lang = 'pl'
}
```